.github/workflows/testing.yml

```diff
@@ -26,42 +31,70 @@ jobs:
       - name: Check resolving dependencies did not change the lock file
         run: |
           echo "Checking if locking dependencies has created any diff in the lock file..."
-          if 
-            git diff --exit-code -- poetry.lock
-          then
-            echo "No changes to poetry.lock detected"
-          else
+          set +e
+          git diff --exit-code -- poetry.lock > /dev/null
+          if [ $? -ne 0 ] ; then
             echo "Changes to poetry.lock detected!"
             echo "You probably should run 'poetry lock --no-update' and add the result to your PR"
+            set -e
             exit 1
           fi
 
+  model-docs:
+    # Job to check that update to models' documentation is not required
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./.github/actions/setup-python-and-dependencies
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+          poetry-version: ${{ env.POETRY_VERSION }}
+      - run: poetry run poe generate-model-docs
+      - name: Check model changes left undocumented
+        run: |
+          echo "Checking if model changes left undocumented"
+          set +e
```

Comment:
unsetting `e` and checking `$?` feels more complex than what we had before, why not simply do
```bash
if
    ! git diff --exit-code -- docs/* > /dev/null
else
    echo "changes..."
    exit 1
fi
```